src/modules/product/components/ProductEditShoppingCart/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Color, Size } from 'types/types';
+import SizeSelector from '../SizeSelector';
+import ColorSelector from '../ColorSelector';
+import ProductPrice from '../ProductPrice';
+import { useAppDispatch, useAppSelector } from 'hooks';
+import {
+  selectStockQuantity,
+  selectQuantityByProductId,
+  actions as shoppingCartActions,
+} from 'redux/slices/shopping-cart';
+import Minus from 'assets/svgs/Minus';
+import Plus from 'assets/svgs/Plus';
+import styles from './index.module.scss';
+
+const MIN_SELECTED_COUNT = 1;
+
+type ProductEditShoppingCartProps = {
+  productId: string;
+  size: Size;
+  colour: Color;
+  count: number;
+  totalPrice: number;
+  onChangeSize: (parameter: string, value: Size) => void;
+  onChangeColor: (parameter: string, value: Color) => void;
+};
+
+const ProductEditShoppingCart: React.FC<ProductEditShoppingCartProps> = ({
+  productId,
+  size,
+  colour,
+  count,
+  totalPrice,
+  onChangeColor,
+  onChangeSize,
+}): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const stockQuantity =
+    useAppSelector(state => selectStockQuantity(state, productId)) ?? 0;
+  const sameQuantityInCart =
+    useAppSelector(state => selectQuantityByProductId(state, productId)) ?? 0;
+  const [selectedCount, setSelectedCount] = useState<number>(count);
+
+  const increaseCount = useCallback(() => {
+    if (stockQuantity - sameQuantityInCart > 0) {
+      setSelectedCount(prevCount => prevCount + 1);
+    }
+  }, [selectedCount, sameQuantityInCart, stockQuantity]);
+
+  const decreaseCount = useCallback(() => {
+    if (selectedCount > MIN_SELECTED_COUNT) {
+      setSelectedCount(prevCount => prevCount - 1);
+    }
+  }, [selectedCount]);
+
+  const changeItemAction = useCallback(() => {
+    dispatch(
+      shoppingCartActions.changeItem({
+        id: productId,
+        oldColor: colour,
```

Comment:
The item ID is not a unique identifier because two items with different colors or sizes can have the same ID. Therefore, to identify the item I should include size and color.